hapi-fhir-base/src/test/java/ca/uhn/fhir/rest/param/DateParamTest.java

```
@@ -0,0 +1,74 @@
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DateParamTest {
+
+	@Test
+	public void testBasicParse() {
+		DateParam input = new DateParam("2020-01-01");
+
+		// too bad value is a j.u.Date instead of a new JSR-310 type
+		// DataParam parses using default tz, so go backwards.
+		ZonedDateTime zonedDateTime = input.getValue().toInstant().atZone(ZoneId.systemDefault());
+		assertEquals(2020,zonedDateTime.getYear());
+		assertEquals(Month.JANUARY,zonedDateTime.getMonth());
+		assertEquals(1,zonedDateTime.getDayOfMonth());
+		assertNull(input.getPrefix());
+	}
+
+	@Test
+	public void testBadDateFormat() {
+		try {
+			DateParam input = new DateParam("09-30-1960");
+			fail();
+		} catch (DataFormatException e) {
+			// expected
+		}
+	}
+
+	@Test
+	public void testPrefixParse() {
+		DateParam input = new DateParam("gt2020-01-01");
+
+		assertEquals(ParamPrefixEnum.GREATERTHAN, input.getPrefix());
+	}
+
+	/**
+	 * We support legacy prefixes in addtion to the standard ParamPrefixEnum values.
+	 *
+	 * Testing here since BaseParamWithPrefix is abstract.
+ 	 */
+	@Test
+	public void testLegacyPrefixParse() {
+		assertEquals(ParamPrefixEnum.GREATERTHAN_OR_EQUALS, translateLegacyPrefix(">="));
+		assertEquals(ParamPrefixEnum.GREATERTHAN, translateLegacyPrefix(">"));
+		assertEquals(ParamPrefixEnum.LESSTHAN_OR_EQUALS, translateLegacyPrefix("<="));
+		assertEquals(ParamPrefixEnum.LESSTHAN, translateLegacyPrefix("<"));
+		assertEquals(ParamPrefixEnum.APPROXIMATE, translateLegacyPrefix("~"));
+		assertEquals(ParamPrefixEnum.EQUAL, translateLegacyPrefix("="));
+	}
+
+	private ParamPrefixEnum translateLegacyPrefix(String legacyPrefix) {
+		DateParam input = new DateParam(legacyPrefix + "2020-01-01");
+		return input.getPrefix();
+	}
+
+	@Test
+	public void testJunkDateIssue1465() {
+		// Issue 1464 - the string "junk" wasn't returning 400 as expected.  Parsed as a prefix instead, and discarded.
```

Issue 1464/1465 does not exist in github, probably worth dropping it from the  test name.

---

I created issue #2361. Some invalid date parameters are silently ignored.